src/app/services/tool/tools.service.ts

```diff
@@ -26,14 +26,21 @@ export class ToolsService {
     return this.todayTime = hh + ":" + mn + ":" + ss;
   }
 
-  userName(name: string){
-    let startindex = name.indexOf('<')
-    let endindex = name.indexOf('>')
-    return name.substring(startindex+1,endindex);
+  getEmailString(email: string){
+    let startindex = email.indexOf('<')
```

Comment:
indexof() will not work on undefined values. So first check if email is undefined and return before performing indexOf().